### ISSUE TITLE
JS101 Session 3 May - Hacking on our project with Redux and tests

### ISSUE THREAD

**Issue by katjad** [CodeHubOrg/discussions#22]
Today was all about coding - making tests pass in the new Redux branch of our app (https://github.com/CodeHubOrg/organisations-database/tree/redux) that @rinse0ut had written, to be precise. I think we all enjoyed it so much that we could have gone on for much longer. I for one completely lost track of time. 

We split up in three teams, working on actions, components and reducers. Each team got at least some tests to pass, although we needed some help to get started. Thanks @rinse0ut for the mentoring, and of course for creating the Redux version of the app plus tests!

Minor stumbling blocks before the failing tests could be run, just in case somebody else runs into problems:
- Node dependencies: When running `npm install`, on some machines, there were error messages about unmet `siblings' dependencies`. In at least one case, on a Windows machine, this could be resolved through upgrading Node and npm. 
- Checking out the new branch from the Git repository: If you have previously cloned the repository, running `git pull` will not automatically pull the new branch, only a reference to it. If you type `git branch -a`, all remote branches will be listed, but only those local branches that you had before. To check out the new remote branch locally, run `git checkout -b <newbranch> origin/<newbranch>`. Note: If you had originally forked the repository and later added the original repo as _upstream_, it's _upstream_ instead of _origin_. You can check the remotes in your git configuration with `git remote -v`(see also https://github.com/CodeHubOrg/discussions/issues/9 about keeping your fork in sync with original repo)

/cc @gicela @wingedeel @me--2014 @mattpill @marckimbrey @mjg17 + welcome Oliver!

P.S. Linking again to resources for getting started with Redux https://github.com/CodeHubOrg/discussions/issues/21#issuecomment-212208246

**Comment by Gicela** [comment]
Thanks for the write up, @katjad. Our team learnt a lot from the experience - we had fun 👍 
Thank you @rinse0ut for getting us started. Could we split next session, so we can carry on working on it in teams?

**Comment by rinse0ut** [comment]
I'm glad you enjoyed it.  I was thinking we could also do a group code
review next session.

**Comment by Gicela** [comment]
@rinse0ut A code review is a good idea.

**Comment by me--2014** [comment]
Thanks for the write-up @katjad 

After looking at it again, the problem I had was due to the fact that I didn't have the remote tracking branch on my machine. I had the group repo remote ('upstream') and could see its references, but crucially I hadn't run 'git fetch upstream' to get the new redux remote tracking branch on to my machine. Once I had done that, 'git checkout -b redux upstream/redux' worked correctly.

All that means I now have the tests working properly on my machine, hooray!

**Comment by rinse0ut** [comment]
Well done to our JS101 Simple Web hack team last night! (@katjad, @me--2014 & @rinse0ut).  It was a fun night, with some great people, food and music.

There were some interesting apps, ideas and hardware created!  Some winning apps:

Chill and... : Meet up with random people to chill and eat / football /
netflix
Foodents: Share food and cooking for students
Secret Vibrating Arm Patch Messenger:  For sending secret messages (not
ideal for Airports though!)

We completed an app in three hours!  The app allows you to send Beeps or
Boops cross devices using Socket.io and React.  You can read the code / try
it out below:

https://github.com/CodeHubOrg/beep-boop

I have merged our reducer team code into the Redux branch of the
organsiations-database app.  We got 12/18 tests to pass!  Go teams!

If you have time before the next session it would be great if you could
take a look at the Redux and Beep Boop code in preparation for a group code review next session.

https://github.com/CodeHubOrg/organisations-database/tree/redux

It would also be worth having a look at testing as we need to write more
tests!

**Comment by katjad** [comment]
Yes, interesting experience the hack night. I also learned a bit about how not to approach such a thing, but it was all good. 

Just came across this on reddit and thought it was quite interesting, though have not yet looked at the code. I have always liked the idea to have a React component just as part of a web page. And it's a messaging app :) https://medium.com/@sevenleaps/using-react-components-in-non-react-websites-12fb7d71b63f#.jv71yfnnv

**Comment by rinse0ut** [comment]
Thanks to @me--2014 and @ondenman for their contributions to the organisations database project.  We now have 18/18 tests passing!

**Comment by rinse0ut** [comment]
Essential js links:

https://github.com/ericelliott/essential-javascript-links#essential-javascript-links
